batchflow/models/metrics/regression.py

```diff
@@ -0,0 +1,131 @@
+
+    Available methods are:
+
+    - `None` - no averaging, calculate metrics for each output individually
+    - `mean` - calculate metrics for each output, and take their mean.
+
+    **Metrics**
+    All metrics return:
+
+    - a single value if targets are single output i.e. have shape (n_samples, ).
+    - a single value if targets are multi output i.e have shape (n_samples, n_outputs) and `agg` set to `mean`
+    - a vector with (n_outputs, ) items if targets are multi output and `agg` set to `None`
+    """
+
+    def __init__(self, targets, predictions, weights=None, gap=3, multi=False):
+        super().__init__()
+
+        # if-else block bellow process the case when the inputs and targets are list
+        # of arrays-like. Thats happening when we accumulating targets and predictions in the pipeline
+        # via `.update(V('targets', mode='a'), value)`.
+        # Since we are not interested in across batch aggregation we only need to concatenate this arrays along 0 axis
+        # In this scenario tho we can mess with 2 cases:
+        # 1. targets is a single batch of multioutput targets and has the following structure [[], [], ..].
+        # 2. targets is a list of data come from couple of batches in a single output task
+        # and has the same structure [[], [], ..].
+        # In the first case concatenating such arrays do not have any sence. To separate such cases we introduce
+        # 'multi' argument.
+
+        if np.ndim(targets) == 1 or (np.ndim(targets) == 2 and multi):
+            self.targets = np.array(targets)
+            self.predictions = np.array(predictions)
+        else:
+            self.targets = np.concatenate(targets, axis=0)
+            self.predictions = np.concatenate(predictions, axis=0)
+
+        self.weights = np.array(weights).flatten()
+        self.gap = gap
+        self.multi = multi
+
+        self._agg_fn_dict = {
+            'mean': lambda x: np.mean(x),
+        }
+
+    def __getattr__(self, name):
+        if name == "METRICS_ALIASES":
+            raise AttributeError # See https://nedbatchelder.com/blog/201010/surprising_getattr_recursion.html
+        name = METRICS_ALIASES.get(name, name)
+        return object.__getattribute__(self, name)
+
+    def mean_absolute_error(self):
+        return np.average(np.abs(self.predictions - self.targets), axis=0,
+                          weights=self.weights)
+
+    def mean_squared_error(self):
+        return np.average((self.predictions - self.targets) ** 2, axis=0,
+                          weights=self.weights)
+
+    def median_absilute_error(self):
+        return np.median(np.abs(self.predictions - self.targets))
+
+    def max_error(self):
+        return np.max(np.abs(self.predictions - self.targets))
+
+    def root_mean_square_error(self):
+        return np.sqrt(self.mean_squared_error())
+
+    def r2_score(self):
+        """ r2_score """
```

Comment:
Either expand docs (and add them to the rest of metrics), or remove them altogether